src/Umbraco.Core/Packaging/PackagesRepository.cs

```diff
@@ -414,9 +414,10 @@ private void PackageMacros(PackageDefinition definition, XContainer root)
 
             root.Add(macros);
 
-            // get the partial views for macros and package those
-            IEnumerable<string> views = packagedMacros.Select(x => x.MacroSource).Where(x => x.EndsWith(".cshtml"));
-            PackageMacroPartialViews(views, root);
+            // Get the partial views for macros and package those (exclude views outside of the default directory, e.g. App_Plugins\*\Views)
+            IEnumerable<string> views = packagedMacros.Where(x => x.MacroSource.StartsWith(Constants.SystemDirectories.MacroPartials))
+                .Select(x => x.MacroSource.Substring(Constants.SystemDirectories.MacroPartials.Length).Replace('/', '\\'));
```

Comment:
This does the filtering on and removing of the prefix. It also replaces forward slashes to backward slashes, as the virtual path is constructed differently compared to the normal views, so this aligns everything a bit more (both are handled correctly though).